specify/specify7
# Allow COG selection for interactions

fixes #5422



### Checklist

      and self-explanatory (or properly documented)

### Testing instructions

For the preparation:
For all COs be sure they have preps
- Create RS (RS COs with 1 conso child) with different COs, have at least one of them be a part of a consolidate COG 
- Create RS (RS COs with 0 conso child) with different COs, be sure no COs are part of a conso COG 
- Create RS (RS COGs with 1 conso COG) with different COGs, have at least one of them be a consolidate COG 
- Create RS (RS COGs with 0 conso COG) with different COGs, be sure no COGs are conso COG
- Create a mix RS 

For testing: 

CO not Conso COG child using enter cat num
- Open Loan dialog 
- Enter the cat num of a CO that is NOT a part of conso COG 
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps 

CO Conso COG child using enter cat num
- Open Loan dialog 
- Enter the cat num of a CO that is a part of conso COG 
- Add a prep to the loan 
- [ ] Verify that the prep you selected has been added to the Loan perps + the siblings of that CO

CO RS using choose RS
- Open Loan dialog 
- Choose RS with COs
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps

COG non conso using choose RS
- Open Loan dialog 
- Choose RS with COGs that are not conso
- Add a prep to the loan 
- [ ] Verify that only the prep you selected has been added to the Loan perps

COG conso using choose RS
- Open Loan dialog 
- Choose RS with COGs that are conso
- Add a prep to the loan 
- [ ] Verify that the prep you selected has been added to the Loan perps + all the siblings

COG mix using choose RS
- Open Loan dialog 
- Choose RS with mix COGs
- Add a prep to the loan from the conso COG and a prep from the other type
- [ ] Verify that only the prep you selected from the non conso has been added to the Loan perps + the prep from the conso COG and all its siblings
